drivers: dac: stm32: add dac value range check

jeppenodgaard opened the pull request:
Compare value and DAC resolution in `dac_write_value` and return -ERANGE if the value is above supported resolution.

Commit:
drivers: dac: stm32: add dac value range check

Compare value and DAC resolution in `dac_write_value` and return -EINVAL
if the value is above supported resolution.

Signed-off-by: Jeppe Odgaard <jeppe.odgaard@prevas.dk>

martinjaeger reviewed (changes requested):
Thanks @jeppenodgaard.

Looks good to me. However, there is a typo in the commit message (`adc` instead of `dac`).

jeppenodgaard commented:
> Thanks @jeppenodgaard.
> 
> Looks good to me. However, there is a typo in the commit message (`adc` instead of `dac`).

Thanks! Fixed in PR title and commit message.

martinjaeger reviewed (dismissed):
Thanks. Looks good.

May I ask you to add this return value also to the Doxygen comment for `dac_write_value` in `include/zephyr/drivers/dac.h` in an additional commit as part of this PR?

jeppenodgaard commented:
> Thanks. Looks good.
> 
> May I ask you to add this return value also to the Doxygen comment for `dac_write_value` in `include/zephyr/drivers/dac.h` in an additional commit as part of this PR?

The other drivers return `EINVAL` and it seems be the right thing to return. Are you OK with that change instead?

martinjaeger commented:
> The other drivers return `EINVAL` and it seems be the right thing to return. Are you OK with that change instead?

Oh yes, definitely. I was too quick and just had a look whether the `-ERANGE` is documented. Let's go with `-EINVAL` for consistency.

jeppenodgaard commented:
> > The other drivers return `EINVAL` and it seems be the right thing to return. Are you OK with that change instead?
> 
> Oh yes, definitely. I was too quick and just had a look whether the `-ERANGE` is documented. Let's go with `-EINVAL` for consistency.

Fixed and updated commit message.
